src/vtquery.cpp

```diff
@@ -163,12 +190,97 @@ struct CompareDistance {
     }
 };
 
+std::vector<vtzero::index_value_pair> get_comparison_tags(vtzero::feature f) {
+    std::vector<vtzero::index_value_pair> v;
+    while (auto ii = f.next_property_indexes()) {
+        v.push_back(ii);
+    }
+    return v;
+}
+
+// compares a vector of property tags to determine deduplication of features
+bool compare_comparison_tags(std::vector<vtzero::index_value_pair> a, std::vector<vtzero::index_value_pair> b) {
+    if (a.size() != b.size()) {
+        return false;
+    }
+
+    for (size_t i = 0; i < a.size(); ++i) {
+        // get keys
+        uint32_t a_key = a[i].key().value();
+        uint32_t b_key = b[i].key().value();
+
+        // get values
+        uint32_t a_val = a[i].value().value();
+        uint32_t b_val = b[i].value().value();
+
+        if (a_key != b_key || a_val != b_val) {
+          goto stop;
```

Comment:
To break out of this loop without using `goto` just do:

```c++
if (a_key != b_key || a_val != b_val) {
   return false;
}
```

Reply:
Ah nice, I didn't know you could break out of a loop with a return statement. Thanks!